### ISSUE TITLE
Live Reload for Web Interface

### ISSUE THREAD

**Issue by JimmyBoh** [bluejamesbond/Scribe.js#44]
This might have been a design consideration, but I'd really like a live reload on the web portal. Something as simple as a configurable interval to reload would do the trick, just something that would  allow me to leave the logs up and not have to constantly refresh to view them. It really is a great interface, especially when fullscreen.

**Comment by guillaumewuip** [comment]
Hi @JimmyBoh. You're right, it something that should be done. It's really not complicated (just a js `setInterval` or the angular equivalent). As I'm really busy these days and maybe @bluejamesbond is too, feel free to send a pull request :smiley:

**Comment by JimmyBoh** [comment]
I actually am going to work on this tonight. Any quirks when setting up the dev environment?

**Comment by bluejamesbond** [comment]
Live reload with angular might require some perf optimizations since you don't want to reload each row on update. Instead, you want to insert the missing rows at the bottom or top depending on how you sort it. I would also recommend that the added rows have a highting animation such as [this](http://dabblet.com/gist/2343125).

**Comment by JimmyBoh** [comment]
Agreed. I've implement an service before that uses interval polling to get the latest log entries from a C# WebAPI. Is infinite scrolling a thing, or are all logs loaded at once?

**Comment by bluejamesbond** [comment]
@JimmyBoh I think you should a socket api such as socket.io in order to send data back and forth. no, there is no infinite scrolling concept at this point. However, we would be open to it.

**Comment by JimmyBoh** [comment]
@bluejamesbond Awesome, sounds good to me. I know some people get worried when adding extra libraries.

**Comment by JimmyBoh** [comment]
I'm guessing [this is a merge issue in the `.gitignore`](https://github.com/bluejamesbond/Scribe.js/blob/master/.gitignore#L10)? If so is there a grunt file that I need for development?

**Comment by bluejamesbond** [comment]
@JimmyBoh That seems to be a merge issue. I personally don't use a gruntfile. I can't speak for @guillaumewuip . Can you confirm?

**Comment by guillaumewuip** [comment]
Probably an error coming from my personnal gruntfile, for jshint, guys.

Le lun. 13 juil. 2015 02:37, Mathew Kurian <notifications@github.com> a
écrit :

**Comment by JimmyBoh** [comment]
@guillaumewuip No problem, I was able to get everything up and running, started to integrate Socket.IO into the server. I will be quite busy over the next two weeks but will try to squeeze in some development when I can!

**Comment by bluejamesbond** [comment]
Sounds like a plan!
